data-protocols/dsp/dsp-spi/dsp-catalog-spi/src/main/java/org/eclipse/edc/protocol/dsp/spi/catalog/service/CatalogService.java

```diff
@@ -0,0 +1,23 @@
+/*
+ *  Copyright (c) 2023 Fraunhofer Institute for Software and Systems Engineering
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.protocol.dsp.spi.catalog.service;
+
+import jakarta.json.JsonObject;
+
+public interface CatalogService {
+    
+    JsonObject getCatalog(JsonObject filter);
```

Comment:
This should return a `Catalog`, and the filter should be typed as well (`QuerySpec`?)